src/autodev.ts

```diff
@@ -139,32 +160,28 @@ const merge = async (
   await exec(`git reset origin/${base}`)
   await exec('git add -A')
 
+  const overrideDate = {
+    env: {
+      GIT_COMMITTER_DATE: commitDate,
+      GIT_AUTHOR_DATE: commitDate
+    }
+  }
   const message = `AutoDev Merge\n\nThe following branches have been merged:\n${success
     .map(p => `- ${p.branch}`)
     .join('\n')}`
-  await exec('git commit -m', [message])
+
+  await exec('git commit -m', [message], overrideDate)
+  // replace with graft commit so we can preserve commit parents
+  await exec(
+    `git replace --graft HEAD origin/${base}`,
+    success.map(p => `origin/${p.branch}`),
+    overrideDate
+  )
+  const rev = await execAndSlurp('git rev-parse HEAD')
+  await exec(`git checkout replace/${rev}`)
```

Comment:
I guess this is normal and we can't prevent it? 

> You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.
>
> If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:
>
>  git switch -c <new-branch-name>
>
>Or undo this operation with:
>
>  git switch -
>
>Turn off this advice by setting config variable advice.detachedHead to false

Reply:
And I don't know what I do differently, but when I run `npm run all`, I have differences again?!

Reply:
> I guess this is normal and we can't prevent it?

You mean the message? I don't think so

Reply:
> And I don't know what I do differently, but when I run `npm run all`, I have differences again?!

I guess the result of `npm run all` is not exactly reproducible,

Reply:
I did  a rebase now, removed `yarn.lock`, ran `yarn` again and then reran `npm run all`. Hope this result is now reproducible on your end